### PR TITLE
Improve Sample: java security usage 

### PR DESCRIPTION
**Pull request by nenaraab**

### CODE REVIEW

**Review by liga-oz — CHANGES_REQUESTED** [review]
just rename the argument, everything else is awesome!

File: samples/java-security-usage/src/main/java/com/sap/cloud/security/samples/HelloJavaServlet.java [hunk 1]
```diff
@@ -1,44 +1,41 @@
...
 import javax.servlet.http.HttpServletResponse;
 import java.io.IOException;
 
 @WebServlet(HelloJavaServlet.ENDPOINT)
-@ServletSecurity(@HttpConstraint(rolesAllowed = { "Read" }))
 public class HelloJavaServlet extends HttpServlet {
 	static final String ENDPOINT = "/hello-java-security";
-	private static final long serialVersionUID = 1L;
-	private static Logger logger = LoggerFactory.getLogger(HelloJavaServlet.class);
+	private static Logger LOGGER = LoggerFactory.getLogger(HelloJavaServlet.class);
```
[the remark this answers is not in the document]
Comment (nenaraab) [reply samples/java-security-usage/src/main/java/com/sap/cloud/security/samples/HelloJavaServlet.java:18]: Static logger references should be in Capital cases

**Comment by nenaraab** [comment]
HI @liga-oz can you please perform another review? The parameter change is applied.

File: samples/java-security-usage/src/main/java/com/sap/cloud/security/samples/HelloJavaServlet.java [hunk 2]
```diff
@@ -1,44 +1,43 @@
...
```
Comment (liga-oz) [inline samples/java-security-usage/src/main/java/com/sap/cloud/security/samples/HelloJavaServlet.java:18]: i guess logger could also be final?

File: samples/java-security-usage/src/main/java/com/sap/cloud/security/samples/XsuaaSecurityFilter.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+			try {
+				HttpServletResponse httpServletResponse = (HttpServletResponse) response;
+				httpServletResponse.sendError(HttpServletResponse.SC_UNAUTHORIZED, unauthenticatedReason); // 401
+			} catch (IOException e) {
+				LOGGER.error("Failed to send error response", e);
+			}
+		}
+	}
+
+	public static void sendUnauthorizedResponse(ServletResponse response, String missingScope)  {
```
Comment (liga-oz) [inline samples/java-security-usage/src/main/java/com/sap/cloud/security/samples/XsuaaSecurityFilter.java:58]: missing scope looks really good here

**Review by liga-oz — APPROVED** [review]
@nenaraab all looks good, just please check if the logger in servlet class can also be final.

File: samples/java-security-usage/src/main/java/com/sap/cloud/security/samples/HelloJavaServlet.java (hunk 2 above)
Comment (nenaraab) [reply samples/java-security-usage/src/main/java/com/sap/cloud/security/samples/HelloJavaServlet.java:18]: yes. Good idea.

**Pull request merged**